astropy/table/index.py

```diff
@@ -398,9 +398,8 @@ def __deepcopy__(self, memo):
         ----------
         memo : dict
         '''
-        num_cols = self.data.num_cols if self.engine == SortedArray else None
```

Comment:
Good catch :)

Reply:
The advantage of having automatic checking in my emacs - I see red squiggly lines under definitions of variables that do not get used.

Reply:
Emacs 4ever!